packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,204 @@
+          middleware: [
+            flip(),
+            size({
+              apply: ({ availableHeight, rects }) => {
+                this.menuListContainer?.style &&
+                  Object.assign(this.menuListContainer.style, {
+                    maxHeight: `${availableHeight}px`,
+                    width: `${rects.reference.width}px`,
+                  });
+              },
+            }),
+            hide(),
+          ],
+        });
+        if (middlewareData.hide?.referenceHidden) {
+          this.expanded = false;
+          return;
+        }
+
+        this.menuListContainer?.style &&
+          Object.assign(this.menuListContainer.style, {
+            position: 'fixed',
+            top: '0',
+            left: '0',
+            transform: `translate(${x}px, ${y}px)`,
+          });
+      });
+    }
+  }
+
+  // Declare menu as an observable property of type HTMLElement[]
+  @observable
+  public menu?: HTMLElement[];
+
+  // Method to update the menu when it changes
+  protected menuChanged(): void {
+    if (this.$fastController.isConnected && this.menu && this.menu.length) {
+      this._menu = this.menu[0] as MenuList;
+    }
+  }
+
+  // Declare trigger as an observable property of type HTMLElement[]
+  @observable
+  public trigger?: HTMLElement[];
+
+  // Method to update the trigger when it changes
+  protected triggerChanged(prev: HTMLElement[] | void, next: HTMLElement[]): void {
+    if (prev !== next) {
+      if (Array.isArray(prev) && prev.length) {
+        this.cleanupTriggerEventListeners(prev[0]);
+      }
+
+      if (next && next.length) {
+        this._trigger = next[0];
+
+        this._trigger.setAttribute('aria-haspopup', 'true');
+        this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+
+        // Add new event listeners
+        if (this.openOnHover) {
+          this._trigger.addEventListener('mouseover', this.toggleExpanded);
+        } else {
+          this._trigger.addEventListener('click', this.toggleExpanded);
+        }
+      }
+    }
+  }
+
+  // Declare expanded as an observable boolean property
+  @observable
+  @attr({ mode: 'boolean' })
+  public expanded: boolean = false;
+
+  // Method to update the expanded property when it changes
+  protected expandedChanged(): void {
+    if (this.$fastController.isConnected && this._trigger) {
+      this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+    }
+    Updates.enqueue(() => {
+      this.setPositioning();
+      this.focus();
+    });
+  }
+
+  // Method to focus on the menu when expanded
+  public focus(): void {
+    if (this.expanded && this._menu) {
+      const menu = this._menu as MenuList;
+      Updates.enqueue(() => menu.focus());
+    }
+  }
+
+  // Method to toggle the expanded property
+  protected toggleExpanded = (e: MouseEvent) => {
+    this.expanded = !this.expanded;
+  };
+
+  // Method to close the menu
+  private closeMenu() {
+    this.expanded = false;
```

Comment:
if the menu can close on its own, we should likely have an event emitted to notify that it's happened along with the state and any other relevant information. If the menu can likewise be opened in a controlled and uncontrolled state we should likely do the same.